services/event-application-service/src/pubsub_controller/pubsub_subscribers.h

```diff
@@ -0,0 +1,151 @@
+
+class PubSubController;
+
+// MARK: - Create subs
+
+class CreateFormSub final {
+ public:
+  static constexpr std::string_view topic = "eas_form_create";
+
+  explicit CreateFormSub(std::shared_ptr<commands::ICommandHandler> handler)
+      : handler_{handler} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  std::shared_ptr<commands::ICommandHandler> handler_;
+};
+
+class CreateResponseSub final {
+ public:
+  static constexpr std::string_view topic = "eas_response_create";
+
+  explicit CreateResponseSub(std::shared_ptr<commands::ICommandHandler> handler)
+      : handler_{handler} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  std::shared_ptr<commands::ICommandHandler> handler_;
+};
+
+// MARK: - Get subs
+
+class GetFormSub final {
+ public:
+  static constexpr std::string_view topic = "eas_form_get";
+  explicit GetFormSub(utils::non_owning<PubSubController *> controller,
+                      std::shared_ptr<queries::IQueryHandler> reader)
+      : controller_{controller}, reader_{reader} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  utils::non_owning<PubSubController *> controller_;
+  std::shared_ptr<queries::IQueryHandler> reader_;
+};
+
+class GetResponseSub final {
+ public:
+  static constexpr std::string_view topic = "eas_response_get";
+
+  explicit GetResponseSub(utils::non_owning<PubSubController *> controller,
+                          std::shared_ptr<queries::IQueryHandler> reader)
+      : controller_{controller}, reader_{reader} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  utils::non_owning<PubSubController *> controller_;
+  std::shared_ptr<queries::IQueryHandler> reader_;
+};
+
+class GetAllEventResponses final {
+ public:
+  static constexpr std::string_view topic = "eas_form_responses_get";
+  explicit GetAllEventResponses(
+      utils::non_owning<PubSubController *> controller,
+      std::shared_ptr<queries::IQueryHandler> reader)
+      : controller_{controller}, reader_{reader} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  utils::non_owning<PubSubController *> controller_;
+  std::shared_ptr<queries::IQueryHandler> reader_;
+};
+
+// MARK: - Update subs
+
+class UpdateFormSub final {
+ public:
+  static constexpr std::string_view topic = "eas_form_update";
+
+  explicit UpdateFormSub(std::shared_ptr<commands::ICommandHandler> handler)
+      : handler_{handler} {}
+  void operator()(google::cloud::pubsub::Message const &msg,
+                  google::cloud::pubsub::AckHandler ack);
+
+ private:
+  std::shared_ptr<commands::ICommandHandler> handler_;
+};
```

Comment:
As I can see, we can make two generic interfaces

Reply:
What interfaces?

Reply:
Constructor and operator everywhere are the same, so it could be written once in base class and then inherited